src/extract_weather.py

```diff
@@ -0,0 +1,127 @@
+"""Extract script for the Open-Meteo API"""
+
+from datetime import datetime, UTC
+import openmeteo_requests
+import pandas as pd
+import requests_cache
+from retry_requests import retry
+
+
+def get_client(cache_expiry: int) -> openmeteo_requests.Client:
+    """Returns Open-Meteo requests client"""
+    cache_session = requests_cache.CachedSession(
+        '.cache', expire_after=cache_expiry)
+    retry_session = retry(cache_session, retries=5, backoff_factor=0.2)
+
+    return openmeteo_requests.Client(session=retry_session)
+
+
+def get_response(latitude: float, longitude: float, client: openmeteo_requests.Client):
+    """Returns a response, provided with latitude, longitude, and an Open-Meteo requests client"""
+    url = "https://api.open-meteo.com/v1/forecast"
+    params = {
+        "latitude": latitude,
+        "longitude": longitude,
+        "daily": ["sunrise", "sunset", "wind_speed_10m_max", "wind_gusts_10m_max",
+                  "temperature_2m_max", "temperature_2m_min"],
+        "hourly": ["temperature_2m", "relative_humidity_2m", "precipitation",
+                   "wind_speed_10m", "wind_gusts_10m", "visibility", "cloud_cover",
+                   "surface_pressure", "pressure_msl"],
+        "models": "ukmo_seamless",
+        "current": ["temperature_2m", "relative_humidity_2m", "wind_speed_10m",
+                    "wind_gusts_10m", "precipitation", "cloud_cover"],
+        "timezone": "GMT",
+        "forecast_days": 7,
+    }
+    return client.weather_api(url, params=params)[0]
+
+
+def process_current_data(response):
+    """Prints current weather data from the API response"""
+    current = response.Current()
+    current_temperature_2m = current.Variables(0).Value()
+    current_relative_humidity_2m = current.Variables(1).Value()
+    current_wind_speed_10m = current.Variables(2).Value()
+    current_wind_gusts_10m = current.Variables(3).Value()
+    current_precipitation = current.Variables(4).Value()
+    current_cloud_cover = current.Variables(5).Value()
+
+    # Some of this could be used as metrics on the dashboard
+    print(
+        f"\nCurrent time: {datetime.fromtimestamp(current.Time(),
+                                                  UTC).strftime('%Y-%m-%d %H:%M:%S')}")
```

Comment:
suggestion: this line is quite hard to read - a slight refactor would help a lot
```suggestion

    # Some of this could be used as metrics on the dashboard
    current_time = datetime.fromtimestamp(
        timestamp=current.Time(),
        tz=UTC
    ).strftime('%Y-%m-%d %H:%M:%S')

    print(f"\nCurrent time: {current_time}")
```